feat: build on wasm

Frando opened the pull request:
## Description

Make iroh-docs work in WebAssembly in the browser, memstore only.


* Has the usual changes to use `n0-future` for time and tasks.
* `redb` works in-memory in the browser. No persistent store for now. In native environments, we spawn a separate thread for the storage actor with a single-threaded tokio runtime. In wasm, we instead simply run this on the main thread. With the inmemory store this is fine,  IMO.
* This PR includes a refactor that was needed anyway: A couple of methods on `Replica` are now async because we previously used `send_blocking` on a `async_channel::Sender` (uuh) and `async_channel::Sender::send_blocking` is not available on wasm. 

## Breaking Changes

Made a couple of methods on `Replica` async because we previously used `send_blocking` on a `async_channel::Sender` (uuh) and `async_channel::Sender::send_blocking` is not available on wasm. This is a much needed refactor anyways.

(needs more details)

## Notes & open questions

<!-- Any notes, remarks or open questions you have to make about the PR. -->

## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

Frando committed:
feat: prepare for wasm support

Frando committed:
feat: further wasm preparations

Frando committed:
refactor: build on wasm, do not use send_blocking

Frando committed:
use git dep

Frando committed:
add build.rs

Frando committed:
remove path dep

Frando committed:
more time fixes

Frando committed:
fixup

Frando committed:
fixup

Frando committed:
fixup

Frando committed:
feature gate fs tests

Frando committed:
more feature flags

Frando committed:
more feature flags

Frando committed:
chore: add wasm ci

b5 committed:
chore(deps): fixup deps

b5 committed:
chore(deps): more deps fixes, replace Cargo.lock

b5 commented:
I've taken the liberty of updating this PR, all of the upstream PRs this one depends on are now merged, we're only waiting on a release for n0-future that'll include the features this relies on.

matheus23 commented:
n0-future 0.3.1 is released now :v:

Frando committed:
deps: use n0-future 0.3.1

b5 committed:
fix(wasm): std::time::Duration -> n0_future::time::Duration

b5 committed:
fix(time): replace stray reference to UNIX_EPOCH

b5 committed:
fix(iroh-blobs): use iroh-blobs fixes

robcohen commented:
Excited to see this land\! I'm building a P2P sync library for Obsidian (note-taking app) and have been designing around the iroh-docs API so we can swap it in once WASM support is ready. Looking forward to using native range reconciliation instead of our custom Prolly tree sync. 🎉

cbenhagen committed:
chore: resolve merge conflicts with main (v0.96)

- Update dependency versions to 0.96 era, keep default-features=false
- Drop iroh-blobs git dep, use released v0.98
- Use IntoAddressLookup API (renamed from IntoDiscovery in 0.96)
- Keep main's docs-rs CI approach
- Regenerate Cargo.lock

cbenhagen commented:
@Frando I opened #84 to resolve the merge conflicts with main (v0.96). Hope that helps unblock this PR!

Frando committed:
chore: cleanup

Frando committed:
chore: pin pkarr @ 5.0.2

Frando committed:
fixup

b5 reviewed (approved):
🙌🙌🙌
